extensions/amp-story/1.0/amp-story.js

```diff
@@ -1015,18 +1015,18 @@ export class AmpStory extends AMP.BaseElement {
    */
   initializeLiveStory_() {
     if (this.element.hasAttribute('live-story')) {
-      this.liveStoryManager_ = new LiveStoryManager(this);
+      this.liveStoryManager_ = new LiveStoryManager(this, this.getAmpDoc());
       this.liveStoryManager_.build();
 
       this.storeService_.dispatch(Action.ADD_TO_ACTIONS_WHITELIST, [
         {tagOrTarget: 'AMP-LIVE-LIST', method: 'update'},
       ]);
 
-      this.element.addEventListener(AmpEvents.DOM_UPDATE, ({target}) => {
-        this.liveStoryManager_.update(
-          target,
-          this.element.querySelectorAll('amp-story-page:not([ad])')
-        );
+      this.element.addEventListener(AmpEvents.DOM_UPDATE, () => {
+        this.liveStoryManager_.update().then(() => {
+          this.setDesktopPositionAttributes_(this.activePage_);
+          this.preloadPagesByDistance_();
+        });
```

Comment:
Can we do the `initializePages_()` from this block too, so it remains private? Not a big deal but it might make the code a bit more centralized, which is cool :)

Reply:
Yes. I was testing something and forgot to put it back in amp-story 😅